nilearn/plotting/html_stat_map.py

```diff
@@ -1,148 +1,438 @@
 """
-Visualizing 3D stat maps in a Papaya viewer
+Visualizing 3D stat maps in a Brainsprite viewer
 """
+import warnings
 import os
-import tempfile
-import base64
 import json
+from io import BytesIO
+from base64 import encodebytes
+import numbers
 
 import numpy as np
+import matplotlib.pyplot as plt
+import matplotlib.cm as mpl_cm
+from matplotlib.image import imsave
+from matplotlib import colors
 
-from .. import image, datasets
+from nibabel.affines import apply_affine
+
+from ..image import resample_img, resample_to_img, new_img_like
+from .js_plotting_utils import get_html_template, HTMLDocument
+from ..plotting import cm
+from ..plotting.find_cuts import find_xyz_cut_coords
+from ..plotting.img_plotting import _load_anat, _get_colorbar_and_data_ranges
 from .._utils import check_niimg_3d
 from .._utils.extmath import fast_abs_percentile
-from .._utils.param_validation import check_threshold
-from .js_plotting_utils import get_html_template, HTMLDocument, colorscale
-
-
-def _encode_nii(img):
-    # downcast to float32 to save memory
-    img = image.load_img(img, dtype='auto')
-    fd, temp = tempfile.mkstemp(suffix='.nii')
-    os.close(fd)
-    try:
-        img.to_filename(temp)
-        with open(temp, 'rb') as f:
-            binary = f.read()
-            encoded = base64.b64encode(binary).decode('utf-8')
-            return encoded
-    finally:
-        os.remove(temp)
-
-
-def _decode_nii(encoded):
-    fd, temp = tempfile.mkstemp(suffix='.nii')
-    os.close(fd)
-    try:
-        with open(temp, 'wb') as f:
-            f.write(base64.b64decode(encoded.encode('utf-8')))
-        img = image.load_img(temp)
-        # The asarray/copy is to get rid of memmapping
-        loaded = image.new_img_like(img,
-                                    np.asarray(img.get_data()).copy())
-        del img
-        return loaded
-    finally:
-        os.remove(temp)
+from .._utils.niimg import _safe_get_data
+from ..datasets import load_mni152_template
+
+
+def _resample_to_self(img, interpolation):
+    """ For internal use.
+        Resample an image to have a diagonal, isotropic affine.
+    """
+
+    u, s, vh = np.linalg.svd(img.affine[0:3, 0:3])
+    vsize = np.min(np.abs(s))
+    img = resample_img(img, target_affine=np.diag([vsize, vsize, vsize]),
+                       interpolation=interpolation)
+    return img
+
+
+def _data2sprite(data):
+    """ For internal use.
+        Convert a 3D array into a sprite of sagittal slices.
```

Comment:
Maybe a bit more detail, so that the later maintainers understand this. For instance, I would suggest giving an idea of the shape of the output sprite (it doesn't need to be the exact shape, just to convey the idea).